index.html

```diff
@@ -857,6 +852,91 @@
+\begin{equation} \label{interfaced_lambertian_albedo}
+E_c(\omega_o) = F(\omega_o, \eta_c) + E_b \Bigl(1 - F(\omega_o, \eta_c)\Bigr) \Delta(E_b, \eta_c)
+\end{equation}
+where $\Delta(E_b, \eta_c)$ is a darkening factor given by
+\begin{equation} \label{general_darkening_formula}
+\Delta(E_b, \eta_c) = \frac{1 - K}{1 - E_b K} \ .
+\end{equation}
+We can thus approximate the required boost factor as the reciprocal of this, i.e. $B_0 \approx \Delta^{-1}$. Note that as $E_b \rightarrow 1$, the darkening factor $\Delta \rightarrow 1$, as required since physically no energy is lost if the base reflects all incident energy (in the clear-coat case). If the base albedo $E_b$ is boosted by $B(\delta)$, then clearly at $\delta=0$ the darkening factor cancels and equation [undarkened_coat_albedo] is satisfied as required.
+
+Here $K \in [0,1]$ is the _internal diffuse reflection coefficient_, which corresponds to the fraction of the energy leaving the base which returns
+to the base due to internal reflection from the coat. Higher values of $K$ lead to more internal reflections, thus more darkening and saturation. In the case of a Lambertian base (which should be a reasonable approximation to the rough metal, dielectric, or diffuse cases), $K = K_r$ where, with relative coat IOR $\eta_c$:
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_rough_base}
+K_r = 1 - \Bigl(1 - E_F(\eta_c)\Bigr)/\eta_c^2 \ .
+\end{equation}
+The albedo $E_F(\eta_c)$ is the _hemispherical_ (or _average_) albedo of the coat Fresnel factor [^avg_fresnel]. If the base has specular reflection (due to a smooth metallic or dielectric interface) then equation [general_darkening_formula] still applies, but the appropriate value of the internal diffuse reflection coefficient $K$ should be closer to:
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_smooth_base}
+K_s = F(\omega_o, \eta_c) \ ,
+\end{equation}
+which becomes exact in the limit of a perfectly smooth metallic or dielectric base. There is significantly less darkening in the case of a smooth base, as the specular reflection from the base is less likely to generate total internal reflection at the coat boundary.
+
+We thus recommend in the general case that the darkening factor of equation [general_darkening_formula] be applied, but taking
+\begin{equation} \label{internal_diffuse_reflection_coefficient_for_general_base}
+K = \mathrm{lerp}(K_s, K_r, r_b)
+\end{equation}
+where $r_b$ is an estimate of the base roughness, a blend between metal $r_m$ and dielectric $r_d$ roughness estimates according to the **`base_metalness`** $\mathtt{M}$:
+\begin{equation} \label{base_roughness_estimate}
+r_b = \mathrm{lerp}(r_d, r_m, \mathtt{M}) \ .
```

Comment:
I would have to think about it more, but on first read something feels potentially slightly redundant here.

This formula is computing the overall roughness $r_b$ by blending two roughnesses based on $M$. Then the overall $K$ is computed by blending together $K_s$ and $K_r$ based on $r_b$. But it seems like $K_s$ must have also taken $M$ into account in order to blend between the dielectric and metallic $F$ in $K_s = F(\omega_o, \eta_c)$.

Intuitively it feels like $M$ should only need to appear once in the formulation, for blending the effective roughness of the other components with the effective roughness of the metallic component. Although then maybe the `specular_roughness` $r$ would need to appear twice instead.

In any case, the high-level approach of computing an effective roughness value makes a lot of sense. (We do something very similar to this in Eclair.)

Reply:
See the point above about the Fresnel factor in $K_s = F(\omega_o, \eta_c)$ referring to the coat dielectric, not the base (dielectric or metal).